evidence/pool.go

```diff
@@ -97,11 +97,13 @@ func (evpool *Pool) Update(block *types.Block, state sm.State) {
 // AddEvidence checks the evidence is valid and adds it to the pool.
 func (evpool *Pool) AddEvidence(evidence types.Evidence) (err error) {
 
-	// TODO: check if we already have evidence for this
-	// validator at this height so we dont get spammed
+	// check if evidence is already stored
+	if evpool.store.IsStored(evidence) {
+		return ErrEvidenceAlreadyStored{evidence}
```

Comment:
why are we wrapping evidence here? shouldn't it be the responsibility of a caller?

Reply:
Yeah, I wasn't too sure if it would be helpful to return the evidence so that the caller knows but I guess as it is an input that we can pass the responsibility to the caller